js/main.js

```diff
@@ -245,101 +245,572 @@
+					}
+				}
+			});
+
+			return attributes;
+		},
+
+		/**
+		 * Get template share options for given share attributes
+		 *
+		 * @param attributes
+		 * @returns {array}
+		 * @private
+		 */
+		_getShareOptionsForAttributes: function(attributes) {
+			var options = [];
+
+			// for each option, determine if share option is enabled based on required attributes
+			for(var optionIndex = 0; optionIndex < this._shareOptions.length; optionIndex++) {
+				var shareOption = this._shareOptions[optionIndex];
+
+				var shareOptionAttribute = shareOption.attribute;
+
+				// determine if share option is enabled
+				// looping over required and current attributes
+				var enabled = null;
+				if (attributes) {
+					for(var attrIndex = 0; attrIndex < attributes.length; attrIndex++) {
+						var currentAttribute = attributes[attrIndex];
+						if (currentAttribute.scope === shareOptionAttribute.scope
+							&& currentAttribute.key === shareOptionAttribute.key) {
+							enabled = currentAttribute.enabled;
+						}
+					}
+				}
+
+				// set option
+				if (enabled !== null) {
+					options.push({
+						name: shareOption.name,
+						label: shareOption.label,
+						div: shareOption.div,
+						checked: enabled
+					});
+				}
+			}
+
+			return options;
+		},
+
+
+		_getAttribute: function(attributes, scope, key) {
+			for(var i in attributes) {
+				if (attributes[i].scope === scope
+					&& attributes[i].key === key
+					&& attributes[i].enabled !== null) {
+					return attributes[i];
+				}
+			}
+
+			return null;
+		},
+
+		_updateAttributes: function(attributes, scope, key, enabled) {
+			var updatedAttributes = [];
+
+			// copy existing scope-key pairs from attributes
+			for(var i in attributes) {
+				if (attributes[i].scope !== scope
+					|| attributes[i].key !== key) {
+					updatedAttributes.push({
+						scope: attributes[i].scope,
+						key: attributes[i].key,
+						enabled: attributes[i].enabled
+					});
+				}
+			}
+
+			// update attributes with scope-key pair to update
+			if (scope && key && enabled !== null) {
+				updatedAttributes.push({
+					scope: scope,
+					key: key,
+					enabled: enabled
+				});
+			}
+
+			return updatedAttributes;
+		},
+
+		_hasPermission: function(permissions, permission) {
+			return (permissions & permission) === permission;
+		},
+
+		_removePermission: function(permissions, permission) {
+			return (permissions & ~permission);
+		},
+
+		/**
+		 * Fill share options template based on supplied data map of {{ data-item }}
+		 * @private
+		 */
+		_template: function (data) {
+			if (!this._shareOptionsTemplate) {
+				this._shareOptionsTemplate = Handlebars.compile(
+					'<div class="onlyOfficeShareOptionsDiv1">' +
+					'{{#each shareOptionsDiv1}}' +
+					'<span class="shareOption">' +
+					'<input id="attr-{{name}}-{{cid}}-{{shareWith}}" type="checkbox" name="{{name}}" class="onlyOfficeShareOption checkbox" {{#if checked}}checked="checked"{{/if}} data-share-id="{{shareId}}"/>' +
```

Comment:
Attributes with custom field have no requirements, you can write any logic you need (it can be also dropdown list if you wish), as soon as you pass the correct class (attribute class) to addShare/updateShare while extending.